### PR TITLE
Create timeline component for About Us Page

### LINKED ISSUE #22: Create Scalable Timeline Component for About Us Page
We should try to see if we can make this timeline component scalable for N number of events

image

### PR DESCRIPTION
**Pull request by tylerlam1**
This resolves #22.

Notes:
- I don't have a heading for this yet because I think it will make more sense to add that once we have the about us intro page on the other side
- Similarly, I didn't create make the component scrollable yet since I don't know the approximate height of the left-side component. [Figma file here](https://www.figma.com/file/BFiDlU8rAGlWdMM0BkiYKm/CTCYYC-Website-Redesign)
- I made it with fixed rem widths - I had difficulty aligning everything with % widths

**Webpage component**
image

**Mobile component**

https://user-images.githubusercontent.com/41935461/184974147-97e0f76e-0833-47bb-a0ef-0a0678db71e6.mp4

### CODE REVIEW

File: pages/who-we-are/index.js [hunk 1]
```diff
@@ -34,6 +35,7 @@ const WhoWeAre = () => {
           ))}
         </div>
       </div>
+      <Timeline />
```
Comment (harsohailB) [inline pages/who-we-are/index.js:38]: I would do something like this for now to get closer to what we want:

```jsx
<div className="h-screen flex">
        <div className="w-1/2" /> // placeholder for left side
        <div className="w-1/2 overflow-y-scroll">
          <Timeline />
        </div>
</div>
```

and then create an issue to work on the placeholder. Also move this block above the rest so its at the top

File: pages/_app.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ function MyApp({ Component, pageProps }) {
       </Head>
       <Navbar />
       <Component {...pageProps} />
+      {/* CONTAINER FOR CONTENT BELOW TO TEST NAVBAR IS IN RIGHT POSITION*/}
```
Comment (harsohailB) [inline pages/_app.js]: Is this change intended?

File: components/Timeline.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/* eslint-disable @next/next/no-img-element */
+import { TIMELINE_DATA } from "../data/timeline";
+
+const TIMELINE_CONTAINER = "bg-white flex flex-col";
+
+const Timeline = () => {
+  const middleLine = <div className="border-l-4 border-[#7055FD] h-30" />;
+
+  const renderTimeline = TIMELINE_DATA.map(({ year, description, smallImage, largeImage }, index) => {
+    if (index % 2 == 0) {
```
Comment (harsohailB) [inline components/Timeline.jsx]: nit: extract into variable `const leftSide = index % 2 === 0`

File: components/Timeline.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      return (
+        <div className="flex flex-row flex-start">
```
Comment (harsohailB) [inline components/Timeline.jsx]: Might be more readable if we breakout these into `LeftSide` and `RightSide` components internal to this file (do not export)

**Review by harsohailB — CHANGES_REQUESTED** [review]
Damn very nice. This seems very hard too do but looks amazing 🚀

File: pages/who-we-are/index.js (hunk 1 above)
Comment (tylerlam1) [reply pages/who-we-are/index.js:38]: I have another PR for the other side coming pretty quick so if its okay, let's skip this and review the next PR.

File: pages/_app.js (hunk 2 above)
Comment (tylerlam1) [reply pages/_app.js]: not at all

File: components/Timeline.jsx (hunk 4 above)
Comment (tylerlam1) [reply components/Timeline.jsx]: Good point

**Review by harsohailB — APPROVED** [review]
